### PR TITLE
Cuda Resize Operator for opset 11.

### PR DESCRIPTION
**Pull request by zhanghuanrong**
Implement resize opertor for opset 11 which is major update of the spec.

### CODE REVIEW

**Comment by ybrnathan** [comment]
@Roli is added to the review. #Closed

**Comment by ybrnathan** [comment]
@Duli is added to the review. #Closed

File: onnxruntime/core/providers/cuda/tensor/resize_impl.cu [hunk 1]
```diff
@@ -3,190 +3,463 @@
...
+    const int rank,
+    CudaKernel::CudaAsyncBuffer<int64_t>& input_shape,
+    CudaKernel::CudaAsyncBuffer<int64_t>& output_shape,
+    CudaKernel::CudaAsyncBuffer<int64_t>& input_strides,
+    CudaKernel::CudaAsyncBuffer<fast_divmod>& output_div_pitches,
+    CudaKernel::CudaAsyncBuffer<float>& scales_vals,
+    CudaKernel::CudaAsyncBuffer<float>& roi_vals,
+    const T* input_data,
+    T* output_data,
+    const size_t N,
```
Comment (ybrnathan) [inline onnxruntime/core/providers/cuda/tensor/resize_impl.cu]: it might be a better idea to keep output_data as the last parameter.

**Comment by ybrnathan** [comment]
@YulongW is added to the review. #Closed

File: onnxruntime/core/providers/cuda/tensor/resize_impl.cu [hunk 2]
```diff
@@ -3,190 +3,463 @@
...
+__device__ int NearestPixel_SIMPLE(float x_original, bool is_down_sampling) {
+  if (is_down_sampling) {
+    return static_cast<int>(ceil(x_original));
+  } else {
+    return static_cast<int>(x_original);
+  }
+}
+
+__device__ int NearestPixel_ROUND_PREFER_FLOOR(float x_original, bool) {
+  if (x_original == static_cast<int>(x_original) + 0.5f) {
```
Comment (yufenglee) [inline onnxruntime/core/providers/cuda/tensor/resize_impl.cu]: It should be static_cast<int>(x_original + 0.5f)

**Pull request merged**